string/string.mbt

```diff
@@ -37,8 +55,62 @@ test "compare" {
   @assertion.assert_eq(1, "abc".compare(""))?
   @assertion.assert_eq(-1, "abcd".compare("abce"))?
   @assertion.assert_eq(1, "abce".compare("abcd"))?
-  @assertion.assert_eq(-1, "c".compare("ab"))?
-  @assertion.assert_eq(1, "ab".compare("c"))?
+  @assertion.assert_eq(1, "c".compare("ab"))?
+  @assertion.assert_eq(-1, "ab".compare("c"))?
+
+  // real world examples
+  @assertion.assert_eq(-1, "Alice".compare("Bob"))?
+  @assertion.assert_eq(-1, "Bob".compare("John"))?
+  @assertion.assert_eq(-1, "Alice".compare("John"))?
+
+  // case-sensitive
+  @assertion.assert_eq(1, "alice".compare("Alice"))?
+  @assertion.assert_eq(1, "alice".compare("Bob"))?
+  @assertion.assert_eq(1, "alice".compare("Zulu"))?
+
+  // correct compare with surrogate pairs
+  // note: U+1F600 = 0xD83D 0xDE00 in UTF16
+  @assertion.assert_eq(-1, "\u{AC00}".compare("\u{FE11}"))?
+  @assertion.assert_eq(-1, "\u{FE11}".compare("\u{1F600}"))?
+  @assertion.assert_eq(-1, "\u{AC00}".compare("\u{1F600}"))?
+}
+
+/// Compare two strings case-insensitive.
+///
+/// String are compared char-by-char by unicode code point value, but ASCII letters 
+/// are compared case-insensitive. 
+pub fn compare_case_insensitive(self : String, other : String) -> Int {
+  // NOTE: use traditional while loops, since we have no way to 
+  // poll 2 iterators (Iter[Char]) simultaneously.
```

Comment:
I think you can implement a `zip` for `Iter`

Reply:
@peter-jerry-ye  I once considered `zip`, but I think I'm not capable to decide how to design `zip` to deal with iterators having different lengths. (rustlang panics, python trimmes to shortest, how do moonbit?)

This may be postponed for further optimizations after the design of `zip`?

Reply:
I think you may give two versions of `zip` and `zip_exn` where the latter fails and it follows the naming convention that we have right now.